internal/common/function/function.go

```diff
@@ -0,0 +1,224 @@
+	oldStatus := nab.Spec.BackupStatus.DeepCopy()
+	oldSpec := nab.Spec.BackupSpec.DeepCopy()
+
+	// Update the status & spec
+	nab.Spec.BackupStatus = &veleroBackup.Status
+	nab.Spec.BackupSpec = &veleroBackup.Spec
+
+	if reflect.DeepEqual(oldStatus, nab.Spec.BackupStatus) && reflect.DeepEqual(oldSpec, nab.Spec.BackupSpec) {
+		// No change, no need to update
+		logger.V(1).Info("NonAdminBackup status and spec is already up to date")
+		return nil
+	}
+
+	if err := r.Update(ctx, nab); err != nil {
+		logger.Error(err, "Failed to update NonAdminBackup")
+		return err
+	}
+
+	return nil
+}
+
+// CheckVeleroBackupLabels return true if Velero Backup object has required Non Admin labels, false otherwise
+func CheckVeleroBackupLabels(backup *velerov1api.Backup) bool {
+	// TODO also need to check for constant.OadpLabel label?
+	labels := backup.GetLabels()
+	value, exists := labels[constant.ManagedByLabel]
+	return exists && value == constant.ManagedByLabelValue
+}
+
+// TODO not used
+
+// GetNonAdminBackupFromVeleroBackup return referenced NonAdminBackup object from Velero Backup object, if no error occurs
+func GetNonAdminBackupFromVeleroBackup(ctx context.Context, clientInstance client.Client, backup *velerov1api.Backup) (*nacv1alpha1.NonAdminBackup, error) {
+	// Check if the backup has the required annotations to identify the associated NonAdminBackup object
+	logger := log.FromContext(ctx)
+
+	annotations := backup.GetAnnotations()
+
+	annotationsStr := fmt.Sprintf("%v", annotations)
+	logger.V(1).Info("Velero Backup Annotations", "annotations", annotationsStr)
+
+	if annotations == nil {
+		return nil, fmt.Errorf("backup has no annotations")
+	}
+
+	nabOriginNamespace, ok := annotations[constant.NabOriginNamespaceAnnotation]
+	if !ok {
+		return nil, fmt.Errorf(requiredAnnotationError, constant.NabOriginNamespaceAnnotation)
+	}
+
+	nabOriginName, ok := annotations[constant.NabOriginNameAnnotation]
+	if !ok {
+		return nil, fmt.Errorf(requiredAnnotationError, constant.NabOriginNameAnnotation)
+	}
+
+	nonAdminBackupKey := types.NamespacedName{
+		Namespace: nabOriginNamespace,
+		Name:      nabOriginName,
+	}
+
+	nonAdminBackup := &nacv1alpha1.NonAdminBackup{}
+	err := clientInstance.Get(ctx, nonAdminBackupKey, nonAdminBackup)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch NonAdminBackup object: %v", err)
+	}
+
+	nabOriginUUID, ok := annotations[constant.NabOriginUUIDAnnotation]
+	if !ok {
+		return nil, fmt.Errorf(requiredAnnotationError, constant.NabOriginUUIDAnnotation)
+	}
+	// Ensure UID matches
+	if nonAdminBackup.ObjectMeta.UID != types.UID(nabOriginUUID) {
+		return nil, fmt.Errorf("UID from annotation does not match UID of fetched NonAdminBackup object")
+	}
+
+	return nonAdminBackup, nil
+}
+
+// TODO import?
+// Similar to as pkg/common/common.go:AppendUniqueKeyTOfTMaps from github.com/openshift/oadp-operator
+func mergeUniqueKeyTOfTMaps[T comparable](userMap ...map[T]T) (map[T]T, error) {
```

Comment:
This function needs more comment to quickly understand if latter maps overrides the previous one or vice versa.

Reply:
done